src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/SchemaGenerator.cs

```diff
@@ -11,29 +12,104 @@ internal sealed class SchemaGenerator
+            JsonSchema systemdConsoleLoggerOptionsSchema = GenerateConsoleLoggerOptionsSchema(systemdConsoleFormatterOptions, ConsoleLoggerFormat.Systemd);
+            JsonSchema defaultConsoleLoggerOptionsSchema = GenerateDefaultConsoleLoggerOptionsSchema(simpleConsoleFormatterOptions);
+
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(jsonConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(simpleConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(systemdConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(defaultConsoleLoggerOptionsSchema);
+
             string schemaPayload = schema.ToJson();
 
             //Normalize newlines embedded into json
             schemaPayload = schemaPayload.Replace(@"\r\n", @"\n", StringComparison.Ordinal);
             return schemaPayload;
         }
+
+        public static JsonSchema GenerateConsoleLoggerOptionsSchema(JsonSchema consoleFormatterOptions, ConsoleLoggerFormat consoleLoggerFormat)
+        {
+            JsonSchema consoleLoggerOptionsSchema = new JsonSchema();
+
+            JsonSchemaProperty formatterNameProperty = new JsonSchemaProperty();
+            JsonSchemaProperty formatterOptionsProperty = new JsonSchemaProperty();
+
+            JsonSchema formatterOptionsSchema = new JsonSchema();
+            formatterOptionsSchema.Reference = consoleFormatterOptions;
+
+            formatterOptionsProperty.Reference = formatterOptionsSchema;
+
+            formatterNameProperty.ExtensionData = new Dictionary<string, object>();
+            formatterNameProperty.ExtensionData.Add("const", consoleLoggerFormat.ToString());
+
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.FormatterName), formatterNameProperty);
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.FormatterOptions), formatterOptionsProperty);
+            consoleLoggerOptionsSchema.RequiredProperties.Add(nameof(ConsoleLoggerOptions.FormatterName));
+
+            return consoleLoggerOptionsSchema;
+        }
+
+        public static JsonSchema GenerateDefaultConsoleLoggerOptionsSchema(JsonSchema consoleFormatterOptions)
+        {
+            JsonSchema consoleLoggerOptionsSchema = new JsonSchema();
+
+            JsonSchemaProperty formatterNameProperty = new JsonSchemaProperty();
+            JsonSchemaProperty formatterOptionsProperty = new JsonSchemaProperty();
+            
+            JsonSchema formatterOptionsSchema = new JsonSchema();
+            formatterOptionsSchema.Reference = consoleFormatterOptions;
+
+            formatterOptionsProperty.Reference = formatterOptionsSchema;
+
+            formatterNameProperty.ExtensionData = new Dictionary<string, object>();
+            formatterNameProperty.ExtensionData.Add("const", JsonObjectType.Null);
```

Comment:
This writes out:
```json
"FormatterName": {
  "default": "Simple",
  "const": 8
},
```

which is not correct. Please remove this extension data.

Reply:
Looks like since I changed that at the same time as setting the formatterNameProperty to "Simple" I assumed it actually had played a role in addressing the issue (when it hadn't). This is reverted to the way it was before using `formatterNameProperty.Type = JsonObjectType.Null`